src/Raven.Server/Documents/Sharding/Handlers/Processors/Streaming/ShardedStreamingHandlerProcessorForGetStreamQuery.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+using Raven.Client.Documents.Commands;
+using Raven.Client.Exceptions.Sharding;
+using Raven.Client.Http;
+using Raven.Server.Documents.Commands.Streaming;
+using Raven.Server.Documents.Handlers.Processors.Streaming;
+using Raven.Server.Documents.Queries;
+using Raven.Server.Documents.Sharding.Handlers.ContinuationTokens;
+using Raven.Server.Documents.Sharding.Operations;
+using Raven.Server.Documents.Sharding.Queries;
+using Raven.Server.Documents.Sharding.Streaming;
+using Raven.Server.NotificationCenter;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Sharding.Handlers.Processors.Streaming
+{
+    internal class ShardedStreamingHandlerProcessorForGetStreamQuery : AbstractStreamingHandlerProcessorForGetStreamQuery<ShardedDatabaseRequestHandler, TransactionOperationContext>
+    {
+        public ShardedStreamingHandlerProcessorForGetStreamQuery([NotNull] ShardedDatabaseRequestHandler requestHandler, HttpMethod method) : base(requestHandler, method)
+        {
+        }
+        
+        protected override async ValueTask ExecuteQueryAndWriteAsync(TransactionOperationContext context, IndexQueryServerSide query, string format, string debug, bool ignoreLimit, StringValues properties, RequestTimeTracker tracker, OperationCancelToken token)
+        {
+            //TODO stav: use the same time limited token for the sharded operation (as the non sharded)?
+            using (context.OpenReadTransaction())
```

Comment:
why do you need to open here a read tx? do you read something from database record?

Reply:
has been removed since